integration-cli/docker_cli_exec_test.go

```diff
@@ -618,8 +617,8 @@ func (s *DockerSuite) TestExecStartFails(c *check.C) {
 	name := "exec-15750"
 	dockerCmd(c, "run", "-d", "--name", name, "busybox", "top")
 	c.Assert(waitRun(name), check.IsNil)
-
 	out, _, err := dockerCmdWithError("exec", name, "no-such-cmd")
-	c.Assert(err, check.NotNil, check.Commentf(out))
-	c.Assert(out, checker.Contains, "executable file not found")
+	if !(err != nil || out == "" || strings.Contains(err.Error(), "Error running command in existing container")) {
+		c.Fatalf("exec with non-existent cmd should have errored out with no output")
```

Comment:
when i run these 2 cmds:
docker run -d  --name exec-15750 busybox top
docker exec exec-15750 no-such-cmd
this spits out:
exec: "no-such-cmd": executable file not found in $PATH
but with dockerCmdWithError("exec", name, "no-such-cmd"), it's the err that contains "executable file not found in $PATH"  and out is ""
exitCode not checked because the container "exec-15750" still running...so it's 255 from the contained process
I may be confused, I've been staring at this for a long time...